src/main/kotlin/org/jetbrains/tinygoplugin/heapAllocations/toolWindow/TinyGoHeapAllocsWindow.kt

```diff
@@ -1,88 +1,91 @@
 package org.jetbrains.tinygoplugin.heapAllocations.toolWindow
 
+import com.intellij.ide.impl.ContentManagerWatcher
 import com.intellij.openapi.Disposable
-import com.intellij.openapi.actionSystem.ActionGroup
-import com.intellij.openapi.actionSystem.ActionManager
-import com.intellij.openapi.actionSystem.ActionToolbar
-import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.wm.RegisterToolWindowTask
 import com.intellij.openapi.wm.ToolWindow
-import com.intellij.openapi.wm.ToolWindowFactory
+import com.intellij.openapi.wm.ToolWindowAnchor
+import com.intellij.openapi.wm.ToolWindowManager
 import com.intellij.ui.ScrollPaneFactory
-import com.intellij.ui.border.CustomLineBorder
 import com.intellij.ui.content.Content
-import com.intellij.ui.content.ContentFactory
+import com.intellij.ui.content.ContentManager
 import com.intellij.ui.treeStructure.Tree
-import com.intellij.util.ui.JBUI
-import com.intellij.util.ui.UIUtil
-import org.jetbrains.tinygoplugin.configuration.TinyGoConfiguration
-import org.jetbrains.tinygoplugin.heapAllocations.HeapAllocsWatcher
-import org.jetbrains.tinygoplugin.heapAllocations.TinyGoHeapAllocsSupplier
+import org.jetbrains.tinygoplugin.TinyGoBundle
+import org.jetbrains.tinygoplugin.heapAllocations.TinyGoHeapAlloc
 import org.jetbrains.tinygoplugin.heapAllocations.toolWindow.model.RootNode
 import org.jetbrains.tinygoplugin.heapAllocations.toolWindow.model.TinyGoHeapAllocsTreeModel
+import org.jetbrains.tinygoplugin.icon.TinyGoPluginIcons
 import java.awt.BorderLayout
 import javax.swing.JPanel
-import javax.swing.SwingConstants
 import javax.swing.tree.TreeSelectionModel
 
-class TinyGoHeapAllocationsWindowFactory : ToolWindowFactory {
-    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
-        val myToolWindow = TinyGoHeapAllocsWindow(project)
-        val contentFactory = ContentFactory.SERVICE.getInstance()
-        val content: Content = contentFactory.createContent(myToolWindow.panel, "", false)
-        toolWindow.contentManager.addContent(content)
+private const val TINYGO_HEAP_ALLOC_TOOLWINDOW_ID = "HeapAlloc"
+private const val TINYGO_HEAP_ALLOC_TOOLWINDOW_TITLE = "heapAllocs.view.title"
+
+@Service
+class TinyGoHeapAllocsViewManager(val project: Project) {
+    private lateinit var contentManager: ContentManager
+
+    init {
+        ApplicationManager.getApplication().invokeLater {
+            val registerToolWindowTask: RegisterToolWindowTask =
+                RegisterToolWindowTask.closableSecondary(
```

Comment:
This method won't be available in 2022.1 (221.5080-EAP-CANDIDATE-SNAPSHOT).

Reply:
> In general, it's recommended to use `plugin.xml` to register a tool window. Why did you decide to use another approach?

I decided to use this approach because I found [this fragment](https://github.com/JetBrains/intellij-community/blob/master/plugins/coverage-common/src/com/intellij/coverage/view/CoverageViewManager.java#L48) in the CoverageViewManager class as I used the coverage plugin as a reference. But as a consumer (without downloaded sources) I couldn't see that this API is for internal usage only. You are right, this window must be declared in `plugin.xml` then.

Reply:
In my recent commit [aea9097](https://github.com/JetBrains/tinygo-plugin/pull/7/commits/aea90972408e97b12e4e371e9748fbbcdef1c97c) I have registered the tool window in the plugin.xml file with preserving of new logic.